### PR TITLE
Add Credentials-Project assignment resource

### PR DESCRIPTION
**Pull request by roni-frantchi**
Following #48, adding a resource that will allow to assign the creds to projects

### CODE REVIEW

**Comment by roni-frantchi** [comment]
[TF Provider - Cloud Credentials Project Assignment](https://trello.com/c/yeql8fib/3370-tf-provider-cloud-credentials-project-assignment)

File: client/cloud_credentials_project_assignment.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package client
+
+import "errors"
+
+func (self *ApiClient) AssignCloudCredentialsToProject(projectId string, payload CloudCredentialsProjectAssignmentPatchPayload) (CloudCredentialsProjectAssignment, error) {
+	var result CloudCredentialsProjectAssignment
+	if payload.CredentialIds == nil || len(payload.CredentialIds) == 0 {
+		return result, errors.New("Must specify cloud credentials to assign to be assigned to project")
+	}
+	err := self.http.Patch("/credentials/deployment/project/"+projectId, payload, &result)
```
Comment (eranelbaz) [inline client/cloud_credentials_project_assignment.go:10]: Shouldn't that be `put`?

**Comment by eranelbaz** [comment]
@roni-frantchi please pull from `main`

File: tests/integration/008_cloud_credentials_project_assignment/conf.tf [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  required_providers {
+    env0 = {
+      source = "terraform-registry.env0.com/env0/env0"
+    }
+  }
+}
+
+provider "env0" {}
+
+variable "second_run" {}
```
Comment (avnerenv0) [inline tests/integration/008_cloud_credentials_project_assignment/conf.tf:13]: `second_run` is not needed here ?

**Review by avnerenv0 — APPROVED** [review]
<div><br/>via <a href="https://giphy.com/spongebob/">SpongeBob SquarePants</a> on <a href="https://giphy.com/gifs/spongebob-spongebob-squarepants-season-6-l4FAZDmoWgWhyJpXW">GIPHY</a></div>

File: tests/integration/008_cloud_credentials_project_assignment/conf.tf (hunk 2 above)
Comment (roni-frantchi) [reply tests/integration/008_cloud_credentials_project_assignment/conf.tf:13]: I... don't think so

**Pull request merged**